login/src/main/java/com/glaum/login/controller/CustomerController.java

```diff
@@ -1,25 +1,75 @@
 package com.glaum.login.controller;
 
-import com.glaum.login.model.User;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.glaum.login.configuration.Authorized;
+import com.glaum.login.entity.permission;
 
 import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
 @Controller
 public class CustomerController {
 
+	@Autowired
+	HttpSession httpsessionobj;
+	
+	
+	
     private static final Logger LOGGER = LoggerFactory.getLogger(CustomerController.class);
-
+    
+    //@PreAuthorize("hasPermission('customer', #httpSession)")
     @RequestMapping(value={"/customer"}, method = RequestMethod.GET)
-    public String customer() throws IOException {
+    @Authorized(keys = "view_dashboard")
+    public String customer(HttpSession httpSession,HttpServletRequest request,HttpServletResponse res) throws IOException {
```

Comment:
is it mandatory to have these parameters in the method?
`HttpSession httpSession,HttpServletRequest request,HttpServletResponse res`